src/integrationTest/java/uk/gov/hmcts/reform/bulkscan/payment/processor/client/payhub/PayHubClientPutPaymentTest.java

```diff
@@ -0,0 +1,116 @@
+import uk.gov.hmcts.reform.bulkscan.payment.processor.config.IntegrationTest;
+
+import java.util.function.Function;
+
+import static com.github.tomakehurst.wiremock.client.WireMock.aResponse;
+import static com.github.tomakehurst.wiremock.client.WireMock.equalTo;
+import static com.github.tomakehurst.wiremock.client.WireMock.equalToJson;
+import static com.github.tomakehurst.wiremock.client.WireMock.okJson;
+import static com.github.tomakehurst.wiremock.client.WireMock.put;
+import static com.github.tomakehurst.wiremock.client.WireMock.stubFor;
+import static com.github.tomakehurst.wiremock.client.WireMock.urlEqualTo;
+import static java.util.UUID.randomUUID;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.catchThrowable;
+
+@IntegrationTest
+public class PayHubClientPutPaymentTest {
+
+    private static final String CASE_REF_REQUEST_JSON = "{ \"ccd_case_number\": \"12321321\"}";
+
+    @Autowired
+    private PayHubClient client;
+
+    @Test
+    public void should_return_ok_when_everything_is_ok_ignore_response_body() {
+        // given
+        String s2sToken = randomUUID().toString();
+
+        stubWithRequestAndResponse(
+            s2sToken,
+            "98765342",
+            CASE_REF_REQUEST_JSON,
+            okJson("{\"a\":\"1\"}")
+        );
+
+        // when
+        ResponseEntity updateResponse =
+            client.updateCaseReference(
+                s2sToken,
+                "98765342",
+                new CaseReferenceRequest("12321321")
+            );
+
+        // then
+        assertThat(updateResponse.getStatusCodeValue()).isEqualTo(200);
+
+        //Ignore response body for now !!!
+        assertThat(updateResponse.getBody()).isNull();
+    }
+
+    @ParameterizedTest
+    @EnumSource(value = HttpStatus.class, names = {"BAD_REQUEST", "NOT_FOUND", "INTERNAL_SERVER_ERROR"})
+    public void should_return_PayHubClientException_for_errors(HttpStatus httpStatus) {
+
+        String s2sToken = randomUUID().toString();
+        stubWithRequestAndResponse(
+            s2sToken,
+            "exception_2132131",
+            CASE_REF_REQUEST_JSON,
+            createErrorResponse(b -> b.withStatus(httpStatus.value()))
+        );
+
+        // when
+        Throwable throwable = catchThrowable(() ->
+                                                 client.updateCaseReference(
+                                                     s2sToken,
+                                                     "exception_2132131",
+                                                     new CaseReferenceRequest("12321321")
+                                                 )
+        );
+
+        // then
+        assertThat(throwable).isInstanceOf(PayHubClientException.class);
+
+        // and
+        PayHubClientException exception = (PayHubClientException) throwable;
+
+        assertThat(exception).isNotNull();
```

Comment:
This line should go first in the form:
assertThat(throwable).isNotNull();

Reply:
removed